lms/djangoapps/grades/models.py

```diff
@@ -681,6 +696,18 @@ def _cache_key(cls, course_id):
     def _emit_grade_calculated_event(grade):
         events.course_grade_calculated(grade)
 
+    @classmethod
+    def clear_grade(cls, course_id, user_id):
```

Comment:
I'd suggest making these names a bit more descriptive, here maybe something like `delete_course_grade_for_learner`

Also, a learner should only ever have one course grade per course, so I'd update the logic here to be something like:
try:
 cls.objects.get()
 delete
 get_cache(cls._CACHE_NAMESPACE)[cls._cache_key(course_id)].pop(user_id)
except NotFound, keyerror:
  pass